src/Compilers/VisualBasic/Portable/Binding/Binder_ObjectInitializer.vb

```diff
@@ -508,6 +516,62 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return resultExpression
         End Function
 
+        Private Shared Sub CheckRequiredMembersInObjectInitializer(
+            constructor As MethodSymbol,
+            initializers As ImmutableArray(Of BoundExpression),
+            creationSyntax As SyntaxNode,
+            diagnostics As BindingDiagnosticBag)
+
+            If constructor Is Nothing OrElse constructor.HasSetsRequiredMembers Then
```

Comment:
>constructor Is Nothing [](http://example.com/codeflow?start=15&length=22)

In what situation this condition is true?

Reply:
Looks like this occurs when we see a struct with a `Protected` parameterless ctor, which we would not handle correctly here. I'll add a test and figure out what to do for the scenario.